Sources/BrowserServicesKit/SecureVault/SecureVaultManager.swift

```diff
@@ -158,5 +150,64 @@ extension SecureVaultManager: AutofillSecureVaultDelegate {
             completionHandler(nil)
         }
     }
+    
+    func existingEntries(for domain: String, autofillData: AutofillUserScript.DetectedAutofillData) throws -> AutofillData {
+        
+        let vault = try SecureVaultFactory.default.makeVault(errorReporter: self.delegate)
+        
+        // Determine if the identity should be sent to the client app:
+
+        var proposedIdentity: SecureVaultModels.Identity?
```

Comment:
I know this isn't groundbreaking, but to increase readability you could extract the code responsible for retrieving proposed identity, credentials and credit card into 3 helper methods. It would make this function less scary at first sight :) I believe it's also the right thing to have in case we'll have more autofill types going forward.